scripts/alias_index.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+
+# This script will get all the ElasticSearch indexes that uses a given alias, the alias will
+# be removed from those indexes, before the index name that is passed into this script
+# is aliased using the given alias.
+# e.g. sh ./scripts/alias_index.sh localhost 9200 bi-dev-01082018 bi-dev
+# - The indexes aliased to 'bi-dev' will have their aliases removed (if any exist).
+# - The passed in index name will be aliased using the given alias ('bi-dev').
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-aliases.html#indices-aliases
+
+SCRIPT_NAME=$0
+REQUIRED_NUM_ARGS=4
+
+# Fail fast if we get any errors
+set -e
+
+usage() {
+    echo "usage: ${SCRIPT_NAME} host port index_to_alias alias"
+    echo "  host                    elasticsearch hostname"
+    echo "  port                    elasticsearch port"
+    echo "  index_to_alias          name of new elasticsearch index to alias"
+    echo "  alias                   alias"
+    exit 1
+}
+
+# Fail the script if we recieve an incorrect number of arguments
+if [ $# -ne $REQUIRED_NUM_ARGS ] ; then
```

Comment:
I got this confused in previous reviews. You should use the new bash builtin test operator (`[[ ... ]]`) rather than the old single square bracket test operator or explicit call to `test`.